modules/navier_stokes/src/ics/NSInitialCondition.C

```diff
@@ -24,8 +24,16 @@ NSInitialCondition::validParams()
 {
   InputParameters params = InitialCondition::validParams();
   params.addClassDescription("NSInitialCondition sets intial constant values for all variables.");
-  params.addParam<std::string>(
-      "pressure_variable_name", NS::pressure, "The name of the pressure variable");
+  params.addDeprecatedParam<std::string>("pressure_variable_name",
+                                         NS::pressure,
+                                         "The name of the pressure variable",
+                                         "pressure_variable_name is deprecated, use variable_type");
+  MooseEnum variable_types(
+      "ht e Mach pressure rho rhou rhov rhow rho_et specific_volume temperature vel_x vel_y vel_z");
```

Comment:
You should initialize these with the same `NS` namespace names that are used for comparison in the `value` method